File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        Stack<Character> stack = new Stack();
        char[] arr = s.toCharArray();
        
        for (char c : arr) {
            if (stack.isEmpty()) stack.push(c);
            else if (stack.peek() == c) stack.pop();
            else stack.push(c);
        }
        if (!stack.isEmpty()) return 0;
        else return 1;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    let stack = [];

    for (let c of s) {
        if (stack.length === 0) {
            stack.push(c);
            continue;
        }

        if (stack[stack.length - 1] !== c) {
            stack.push(c);
            continue;
        }
        
        while (stack.length > 0 && stack[stack.length - 1] === c) {
            stack.pop();
        }
    }

    return stack.length === 0 ? 1 : 0;
}